folly/Traits.h

```diff
@@ -312,6 +312,36 @@ struct is_negative_impl<T, false> {
   constexpr static bool check(T x) { return false; }
 };
 
+// Early versions of MSVC choke on the template
+// constraints.
+#if _MSC_FULL_VER <= 200022816 // 2015 RC or below
```

Comment:
AIUI, 2015 final doesn't need this, yes?  If so, we can drop this PR entirely as we're going to declare 2015 final as the minimum version for folly.